### PR TITLE
enh(UI): adding cypress tests for local authentication

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

Adding new cypress scenario for local authentication and refacto some cypress commands. 

## Type of change

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [X] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

Launch cypress test on local authentication.

#### Community contributors & Centreon team

- [X] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/tests/e2e/cypress/integration/Local-authentication/01-local-authentication.feature [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    Then a default password policy and default excluded users must be preset
+    
+Scenario: Enforcing a password case policy
+    Given an administrator configuring a Centreon platform and an existing user account
+    When the administrator sets a valid password length and a sets all the letter cases
+    Then the existing user can not define a password that does not match the password case policy defined by the administrator and is notified about it
+    
+Scenario: Enforcing a password expiration policy
+Scenario: Enforcing a password blocking policy
+Scenario: Enforcing
```
Comment (bdauria) [inline centreon/tests/e2e/cypress/integration/Local-authentication/01-local-authentication.feature:18]: Oops something went wrong

**Review by bdauria — COMMENTED** [review]
Delete the corresponding behat tests (and update the behat.yml)

File: centreon/tests/e2e/cypress/integration/Local-authentication/01-local-authentication.feature (hunk 1 above)
Comment (Thebarda) [reply centreon/tests/e2e/cypress/integration/Local-authentication/01-local-authentication.feature:18]: It's not finished yet

File: centreon/packages/ui/src/Form/FormButtons.tsx [hunk 2]
```diff
@@ -58,13 +58,16 @@ const FormButtons = (): JSX.Element => {
       <div className={classes.buttons}>
         <Button
           aria-label={t(labelReset)}
+          data-testid={labelReset}
           disabled={not(canReset)}
+          id={labelReset?.replace(/[^A-Z0-9]+/gi, '')}
```
Comment (bdauria) [inline centreon/packages/ui/src/Form/FormButtons.tsx:63]: Why is that needed? can we create a util function in order to reuse the id normalization?

Comment (Thebarda) [reply centreon/packages/ui/src/Form/FormButtons.tsx:63]: @SoldierBoy31

Comment (SoldierBoy31) [reply centreon/packages/ui/src/Form/FormButtons.tsx:63]: Agree, I will create it in utils folder but what type of function should i use "getNormalizedId" or useNormalizedId ?

Comment (Thebarda) [reply centreon/packages/ui/src/Form/FormButtons.tsx:63]: If you want a function, pick up the "getNormalizedId". If you want a custom hook, pick the other one

File: centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature [hunk 3]
```diff
@@ -0,0 +1,37 @@
+Feature: Local authentication
+    As a user
+    I want to be able to manage password security policies on a Centreon platform for users going through local authentication
+    So that platform administrators can rely on better password practices and and increased security
+    
+Scenario: Default password policy
+    Given an administrator deploying a new Centreon platform
+    When the administrator opens the authentication configuration menu
+    Then a default password policy and default excluded users must be preset
```
Comment (bdauria) [inline centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature:9]:
```suggestion
    Then a default password policy and default excluded users must be present
```

File: centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+Scenario: Enforcing a password case policy
+    Given an administrator configuring a Centreon platform and an existing user account
+    When the administrator sets a valid password length and a sets all the letter cases
```
Comment (bdauria) [inline centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature:13]:
```suggestion
    When the administrator sets a valid password length and sets all the letter cases
```

File: centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    Then the existing user can not define a password that does not match the password case policy defined by the administrator and is notified about it
+
+Scenario: Enforcing a password expiration policy
+    Given an administrator configuring a Centreon platform and an existing user account with password up to date
+    When the administrator sets valid password expiration policy durations in password expiration policy configuration and the user password expires
+    Then the existing user can not authenticate and is notified about it
+    
+Scenario: Enforcing password change policy
+    Given an administrator configuring a Centreon platform and an existing user account with a first password
+    When the administrator enables the password reuseability and sets and a user attempts to change its password multiple times in a row
```
Comment (bdauria) [inline centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature:23]:
```suggestion
    When the administrator enables the password reuseability and a user attempts to change its password multiple times in a row
```

File: centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    Then user can not change password unless the minimum time has passed
+    Then user can not reuse the last passwords more than 3 times
+    
+Scenario: Editing the excluded users list
+    Given an existing password policy configuration and 2 non admin users
+    When the administrator adds or remove a user from the excluded user list
+    Then the password expiration policy is applied to the user removed
```
Comment (bdauria) [inline centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature:30]:
```suggestion
    Then the password expiration policy is applied to the removed user
```

File: centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    Then the password expiration policy is not applyed anymore to the user added
```
Comment (bdauria) [inline centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature:31]:
```suggestion
    Then the password expiration policy is not applied anymore to the added user
```

File: centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+Scenario: Enforcing a password blocking policy
+    Given an administrator configuring a Centreon platform and an existing user account not blocked
+    When the administrator sets valid password policy blocking and the user attempts to login multiple times
```
Comment (bdauria) [inline centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature:35]:
```suggestion
    When the administrator sets valid password blocking policy and the user attempts to login multiple times
```

File: centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    Then the user is locked after reaching the number of allowed attempts
+    Then the user must wait for the defined duration before attempting again
```
Comment (bdauria) [inline centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature:37]:
```suggestion
    And the user must wait for the defined duration before attempting again
```

File: centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature (hunk 7 above)
Comment (bdauria) [inline centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature:31]:
```suggestion
    And the password expiration policy is not applyed anymore to the user added
```

File: centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature (hunk 9 above)
Comment (SoldierBoy31) [reply centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature:37]: "And" doesn't exist in the new library '@badeball/cypress-cucumber-preprocessor'. Indeed, 'And' and 'But' has been removed since v13.O. Should I keep 'Then'?

File: centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature (hunk 7 above)
Comment (SoldierBoy31) [reply centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature:31]: "And" doesn't exist in the new library '@badeball/cypress-cucumber-preprocessor'. Indeed, 'And' and 'But' has been removed since v13.O. Should I keep 'Then'?

Comment (bdauria) [reply centreon/tests/e2e/cypress/e2e/Local-authentication/01-local-authentication.feature:31]: > "And" doesn't exist in the new library '@badeball/cypress-cucumber-preprocessor'. Indeed, 'And' and 'But' has been removed since v13.O. Should I keep 'Then'?

let's keep the Then, then :)

**Review by bdauria — APPROVED** [review]

File: centreon/tests/e2e/cypress/e2e/Autologin/01-autologin/index.ts [hunk 10]
```diff
@@ -15,11 +15,16 @@ beforeEach(() => {
     method: 'GET',
     url: '/centreon/include/common/userTimezone.php',
   }).as('getTimeZone');
+  cy.intercept({
+    method: 'GET',
+    url: '/centreon/api/latest/users/filters/events-view?page=1&limit=100',
+  }).as('getLastesUserFilters');
```
Comment (Thebarda) [inline centreon/tests/e2e/cypress/e2e/Autologin/01-autologin/index.ts:21]:
```suggestion
  }).as('getLastestUserFilters');
```

File: centreon/tests/e2e/cypress/e2e/Autologin/01-autologin/index.ts [hunk 11]
```diff
@@ -15,11 +15,16 @@ beforeEach(() => {
...
 });
 
 Given('an administrator is logged in the platform', () => {
   cy.loginByTypeOfUser({ jsonName: 'admin', preserveToken: true })
     .wait('@getNavigationList')
+    .wait('@getLastesUserFilters')
```
Comment (Thebarda) [inline centreon/tests/e2e/cypress/e2e/Autologin/01-autologin/index.ts:27]:
```suggestion
    .wait('@getLastestUserFilters')
```

File: centreon/tests/e2e/cypress/support/Commands.ts [hunk 12]
```diff
@@ -112,9 +112,24 @@ Cypress.Commands.add('getIframeBody', (): Cypress.Chainable => {
 });
 
 Cypress.Commands.add(
-  'getFormFieldByIndex',
-  (rootItemNumber: number): Cypress.Chainable => {
-    return cy.getIframeBody().find('#Form').find('tr').eq(rootItemNumber);
+  'handleRequestOnDatabase',
```
Comment (Thebarda) [inline centreon/tests/e2e/cypress/support/Commands.ts:115]: `requestOnDatabase` would be a better name, don't you think?

Comment (SoldierBoy31) [reply centreon/tests/e2e/cypress/support/Commands.ts:115]: What do you think of "runQueryOnDatabaseFromContainer" ?

**Review by SoldierBoy31 — APPROVED** [review]
